castai/resource_workload_scaling_policy_test.go

```diff
@@ -0,0 +1,181 @@
+package castai
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+)
+
+func TestAccResourceWorkloadScalingPolicy(t *testing.T) {
+	rName := fmt.Sprintf("%v-policy-%v", ResourcePrefix, acctest.RandString(8))
+	resourceName := "castai_workload_scaling_policy.test"
+	clusterName := "tf-core-acc-20230723"
+	projectID := os.Getenv("GOOGLE_PROJECT_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckScalingPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: scalingPolicyConfig(clusterName, projectID, rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "apply_type", "IMMEDIATE"),
+					resource.TestCheckResourceAttr(resourceName, "management_option", "READ_ONLY"),
+					resource.TestCheckResourceAttr(resourceName, "cpu.0.function", "QUANTILE"),
+					resource.TestCheckResourceAttr(resourceName, "cpu.0.overhead", "0.05"),
+					resource.TestCheckResourceAttr(resourceName, "cpu.0.apply_threshold", "0.06"),
+					resource.TestCheckResourceAttr(resourceName, "cpu.0.args.0", "0.86"),
+					resource.TestCheckResourceAttr(resourceName, "memory.0.function", "MAX"),
+					resource.TestCheckResourceAttr(resourceName, "memory.0.overhead", "0.25"),
+					resource.TestCheckResourceAttr(resourceName, "memory.0.apply_threshold", "0.1"),
+					resource.TestCheckResourceAttr(resourceName, "memory.0.args.#", "0"),
+				),
+			},
+			{
+				Config: scalingPolicyConfigUpdated(clusterName, projectID, rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", rName+"-updated"),
+					resource.TestCheckResourceAttr(resourceName, "apply_type", "IMMEDIATE"),
+					resource.TestCheckResourceAttr(resourceName, "management_option", "MANAGED"),
+					resource.TestCheckResourceAttr(resourceName, "cpu.0.function", "QUANTILE"),
+					resource.TestCheckResourceAttr(resourceName, "cpu.0.overhead", "0.15"),
+					resource.TestCheckResourceAttr(resourceName, "cpu.0.apply_threshold", "0.1"),
+					resource.TestCheckResourceAttr(resourceName, "cpu.0.args.0", "0.9"),
+					resource.TestCheckResourceAttr(resourceName, "memory.0.function", "QUANTILE"),
+					resource.TestCheckResourceAttr(resourceName, "memory.0.overhead", "0.35"),
+					resource.TestCheckResourceAttr(resourceName, "memory.0.apply_threshold", "0.2"),
+					resource.TestCheckResourceAttr(resourceName, "memory.0.args.0", "0.9"),
+				),
+			},
```

Comment:
Could you please add one more TestStep that verifying that Importer works correctly?